src/repl-frontend/scala/tools/nsc/interpreter/shell/InteractiveReader.scala

```diff
@@ -117,31 +115,7 @@ class SplashLoop(reader: InteractiveReader, prompt: String) extends Runnable {
     thread = null
   }
 
-  /** Block for the result line, or null on ctrl-D. */
-  def line: String = result.take getOrElse null
-}
-object SplashLoop {
-  def apply(reader: SplashReader, prompt: String): SplashLoop = new SplashLoop(reader, prompt)
+  /** Blocking. Returns Some(input) when received during splash loop, or None if interrupted (e.g., ctrl-D). */
+  def line: Option[String] = result.take
 }
 
-/** Reader during splash. Handles splash-completion with a stub, otherwise delegates. */
-class SplashReader(reader: InteractiveReader, postIniter: InteractiveReader => Unit) extends InteractiveReader {
-  /** Invoke the postInit action with the underlying reader. */
-  override def postInit(): Unit = postIniter(reader)
-
-  override val interactive: Boolean = reader.interactive
-
-  override def reset(): Unit = reader.reset()
-  override def history: History = reader.history
-  override val completion: Completion = NoCompletion
-  override def redrawLine(): Unit = reader.redrawLine()
-
-  override protected[interpreter] def readOneLine(prompt: String): String = ???   // unused
-  override protected[interpreter] def readOneKey(prompt: String): Int     = ???   // unused
-
-  override def readLine(prompt: String): String = reader.readLine(prompt)
-}
-object SplashReader {
-  def apply(reader: InteractiveReader)(postIniter: InteractiveReader => Unit) =
-    new SplashReader(reader, postIniter)
-}
```

Comment:
One annoying (to me) behavior is that trying tab completion on first line is thrown away. I'd like it to wait for a PC and then autocomplete. Otherwise I have to hit tab for a couple of seconds (feels like ten). I've gotten into the habit of entering a dummy value just to wait for warm compiler.